mobile/test/swift/integration/ResetConnectivityStateTest.swift

```diff
@@ -11,101 +12,82 @@ final class ResetConnectivityStateTest: XCTestCase {
   }
 
   func testResetConnectivityState() {
-    // swiftlint:disable:next line_length
-    let emhcmType = "type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.EnvoyMobileHttpConnectionManager"
-    // swiftlint:disable:next line_length
-    let assertionFilterType = "type.googleapis.com/envoymobile.extensions.filters.http.assertion.Assertion"
-    let config =
-"""
-listener_manager:
-    name: envoy.listener_manager_impl.api
-    typed_config:
-      "@type": type.googleapis.com/envoy.config.listener.v3.ApiListenerManager
-static_resources:
-  listeners:
-  - name: base_api_listener
-    address:
-      socket_address:
-        protocol: TCP
-        address: 0.0.0.0
-        port_value: 10000
-    api_listener:
-      api_listener:
-        "@type": \(emhcmType)
-        config:
-          stat_prefix: hcm
-          route_config:
-            name: api_router
-            virtual_hosts:
-              - name: api
-                domains:
-                  - "*"
-                routes:
-                  - match:
-                      prefix: "/"
-                    direct_response:
-                      status: 200
-          http_filters:
-            - name: envoy.filters.http.assertion
-              typed_config:
-                "@type": \(assertionFilterType)
-                match_config:
-                  http_request_headers_match:
-                    headers:
-                      - name: ":authority"
-                        exact_match: example.com
-            - name: envoy.router
-              typed_config:
-                "@type": type.googleapis.com/envoy.extensions.filters.http.router.v3.Router
-"""
-    let engine = EngineBuilder(yaml: config)
+    EnvoyTestServer.startHttp1PlaintextServer()
+
+    let engine = EngineBuilder()
       .addLogLevel(.debug)
       .build()
 
     let client = engine.streamClient()
 
-    let requestHeaders = RequestHeadersBuilder(method: .get, scheme: "https",
-                                               authority: "example.com", path: "/test")
+    let port = String(EnvoyTestServer.getEnvoyPort())
+    let requestHeaders = RequestHeadersBuilder(method: .get, scheme: "http",
+                                               authority: "localhost:" + port, path: "/simple.txt")
       .build()
 
     let expectation1 =
       self.expectation(description: "Run called with expected http status first request")
+    let finish1 = self.expectation(description: "endStream sent to callback")
+    var resultEndStream1: Bool = false
 
     client
       .newStreamPrototype()
       .setOnResponseHeaders { responseHeaders, endStream, _ in
          XCTAssertEqual(200, responseHeaders.httpStatus)
```

Comment:
Do you know if Envoy will execute `setOnResponseHeaders` first and then `setOnResponseData`? If for some reason, the `endStream` is set to true on `setOnResponseHeaders` before executing `setOnResponseData`, this test will become flaky. I noticed the same issue when dealing with `SendtestData.kt` in https://github.com/envoyproxy/envoy/pull/33275/files

I think it's probably better to only call `finish1.fullfill()` and `resultEndStream1` inside `setOnResponseData`.

Reply:
In all instances I've seen, Envoy invokes setOnResponseHeaders first, but even if not, I'm not sure I follow the comment, b/c the test only checks if end stream was found in one or the other.  We just want to know we've seen an end stream.  I.e. if we encounter `endStream` on setOnResponseHeaders, then we just won't have anything to do when setOnResponseData gets executed (no expectation to fulfill).

Reply:
Looking at the tests more closely, this is fine given that we have `if endStream` check before calling `finish1.fullfill()`. Sorry for the noise. Resolving.